### DISCUSSION TITLE
NotAllowedError on Chrome for iOS

### DISCUSSION THREAD

**Discussion by ottoptika** [vercel/next.js#35528] [Help]
Hi
I get a "NotAllowedError" when trying to use RecordRTC to record a canvas and download the video. I don't get the error on Brave, Safari or Chrome on my MacBook, and not on Safari and Brave for iOS. Only Chrome for iOS.
Any suggestions to what may cause it? Is it NextJS or RecordRTC or Chrome? Seems like some permission? Can I be sure users have permissions on?
This is the functions I'm trying to call

`const handleRecording = async (e) => {
        e.preventDefault()
        const canvasStream = canvasRef.current.captureStream();
        setStream(canvasStream);
        recorderRef.current = new RecordRTC(canvasStream, { 
            mimeType: 'video/webm;codecs=vp8',
            frameInterval: 25,
            video: HTMLCanvasElement,
        });
        recorderRef.current.startRecording();
      };`

(Sorry about the code-wrapper on GitHub. It doesn't like the formatting)

![IMG_7068](https://user-images.githubusercontent.com/97895941/159514535-b07875ea-63c6-4fa7-89c5-2f0ac4d8de2d.png)
?

**Comment by icyJoseph** [comment]
This is hardly Next.js.

Without code to try and debug this, we can only guess :) 

- Lack of permissions (perhaps you have given permission to localhost:3000 while developing other apps?) Try and change the port, or your local IP, or if you click on the warning sign (triangle with !), it should list what permissions are available.
- This SO thread has two things to try: https://stackoverflow.com/questions/53687528/google-chrome-not-allowing-camera-access-even-when-trusted-and-on-https-site
- Perhaps the feature is not allowed over `HTTP` for Chrome in iOS? Maybe it needs some extra flag? 
- When the component doing this mounts, (or the entire App), run this inside a useEffect:

```ts
navigator.mediaDevices.getUserMedia({video: false, audio: true}).then(console.log)
```

I think it should prompt to ask for user permissions.

**Comment by wpdas** [comment]
iOS Safari needs some more configuration to make it work.

```ts
// Fix for iOS Safari from https://leemartin.dev/hello-webrtc-on-safari-11-e8bcb5335295
video.setAttribute('autoplay', '');
video.setAttribute('muted', '');
video.setAttribute('playsinline', '')
```
